Update OMEVisual to work with the new SCIFIO

```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /Users/curtis/code/fiji-ext/OMEVisual/src/main/java/sc/fiji/omevisual/OMEUtils.java:[17,27] package io.scif.ome.formats does not exist
[INFO] 1 error
[INFO] -------------------------------------------------------------
```

There is no SCIFIO `OMETIFFFormat` anymore. We rely on `scifio-bf-compat` for that. Probably, OMEVisual needs to grab the original format-specific `Metadata` object from the SCIFIO-opened image, and then translate it to an `OMEMetadata`, which it can display.

--- a/pom.xml
+++ b/pom.xml
@@ -112,6 +112,11 @@
 
 		<dependency>
 			<groupId>io.scif</groupId>
+			<artifactId>scifio-bf-compat</artifactId>
+		</dependency>
+
+		<dependency>
+			<groupId>io.scif</groupId>
 			<artifactId>scifio-ome-xml</artifactId>
 		</dependency>
 

--- a/pom.xml
+++ b/pom.xml
@@ -112,12 +112,18 @@
 
 		<dependency>
 			<groupId>io.scif</groupId>
-			<artifactId>scifio-bf-compat</artifactId>
+			<artifactId>scifio-ome-xml</artifactId>
 		</dependency>
 
 		<dependency>
 			<groupId>io.scif</groupId>
-			<artifactId>scifio-ome-xml</artifactId>
+			<artifactId>scifio-bf-compat</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>ome</groupId>
+			<artifactId>bio-formats_plugins</artifactId>
+			<scope>test</scope>
 		</dependency>
 
 		<!-- Enable IJ1 interface at runtime -->
@@ -127,6 +133,7 @@
 			<scope>runtime</scope>
 			<optional>true</optional>
 		</dependency>
+
 
 	</dependencies>
 

--- a/src/main/java/sc/fiji/omevisual/OMEUtils.java
+++ b/src/main/java/sc/fiji/omevisual/OMEUtils.java
@@ -3,19 +3,16 @@
  * To change this template file, choose Tools | Templates
  * and open the template in the editor.
  */
+
 package sc.fiji.omevisual;
+
+import io.scif.Metadata;
+import io.scif.img.SCIFIOImgPlus;
+import io.scif.ome.OMEMetadata;
+import io.scif.services.TranslatorService;
 
 import java.util.Map;
 
-import org.scijava.convert.ConvertService;
-import org.scijava.ui.DialogPrompt;
-
-import ij.ImagePlus;
-import io.scif.Metadata;
-import io.scif.filters.AbstractMetadataWrapper;
-import io.scif.img.SCIFIOImgPlus;
-import io.scif.ome.formats.OMETIFFFormat;
-import loci.formats.ome.OMEXMLMetadata;
 import net.imagej.Dataset;
 import net.imagej.ImageJ;
 import net.imagej.ImgPlus;
@@ -23,13 +20,20 @@ import net.imagej.axis.Axes;
 import net.imagej.axis.AxisType;
 import net.imagej.display.ImageDisplay;
 
+import org.scijava.convert.ConvertService;
+import org.scijava.ui.DialogPrompt;
+
+import ij.ImagePlus;
+import loci.formats.ome.OMEXMLMetadata;
+
 /**
- *
  * @author Hadrien Mary
  */
 public class OMEUtils {
 
-	public static OMEXMLMetadata getOMEXMLMetadata(Dataset data, ImageJ ij) {
+	public static OMEXMLMetadata getOMEXMLMetadata(Dataset data, ImageJ ij,
+		TranslatorService translatorService)
+	{
 
 		ImgPlus<?> imp = data.getImgPlus();
 
@@ -41,22 +45,15 @@ public class OMEUtils {
 
 		SCIFIOImgPlus<?> sciImp = (SCIFIOImgPlus<?>) imp;
 		Metadata metadata = sciImp.getMetadata();
+		OMEMetadata omeMeta = new OMEMetadata(translatorService.getContext());
 
-		// First try translating the given metadata to
-		// Why the fuck this is needed ?
-		while ((metadata instanceof AbstractMetadataWrapper)) {
-			metadata = ((AbstractMetadataWrapper) metadata).unwrap();
-		}
-
-		// Check if metadata are OME or not
-		if (!(metadata instanceof OMETIFFFormat.Metadata)) {
-			ij.ui().showDialog("This file does not contain OME metadata",
+		if (!translatorService.translate(metadata, omeMeta, true)) {
+			ij.ui().showDialog("Unable to extract OME Metadata",
 				DialogPrompt.MessageType.ERROR_MESSAGE);
 			return null;
 		}
 
-		OMETIFFFormat.Metadata omeMeta = ((OMETIFFFormat.Metadata) metadata);
-		OMEXMLMetadata ome = omeMeta.getOmeMeta().getRoot();
+		OMEXMLMetadata ome = omeMeta.getRoot();
 
 		return ome;
 	}

--- a/src/main/java/sc/fiji/omevisual/OMEUtils.java
+++ b/src/main/java/sc/fiji/omevisual/OMEUtils.java
@@ -34,13 +34,15 @@ public class OMEUtils {
 		ImgPlus<?> imp = data.getImgPlus();
 
 		if (!(imp instanceof SCIFIOImgPlus)) {
-			ij.ui().showDialog("This image has not been opened with SCIFIO.", DialogPrompt.MessageType.ERROR_MESSAGE);
+			ij.ui().showDialog("This image has not been opened with SCIFIO.",
+				DialogPrompt.MessageType.ERROR_MESSAGE);
 			return null;
 		}
 
 		SCIFIOImgPlus<?> sciImp = (SCIFIOImgPlus<?>) imp;
 		Metadata metadata = sciImp.getMetadata();
 
+		// First try translating the given metadata to
 		// Why the fuck this is needed ?
 		while ((metadata instanceof AbstractMetadataWrapper)) {
 			metadata = ((AbstractMetadataWrapper) metadata).unwrap();
@@ -48,7 +50,8 @@ public class OMEUtils {
 
 		// Check if metadata are OME or not
 		if (!(metadata instanceof OMETIFFFormat.Metadata)) {
-			ij.ui().showDialog("This file does not contain OME metadata", DialogPrompt.MessageType.ERROR_MESSAGE);
+			ij.ui().showDialog("This file does not contain OME metadata",
+				DialogPrompt.MessageType.ERROR_MESSAGE);
 			return null;
 		}
 
@@ -58,7 +61,9 @@ public class OMEUtils {
 		return ome;
 	}
 
-	public static void setPosition(ImageDisplay image, Map<AxisType, Long> positions, ConvertService convert) {
+	public static void setPosition(ImageDisplay image,
+		Map<AxisType, Long> positions, ConvertService convert)
+	{
 
 		// Work with IJ2 interface
 		positions.forEach((axis, position) -> {

--- a/src/main/java/sc/fiji/omevisual/OMEVisual.java
+++ b/src/main/java/sc/fiji/omevisual/OMEVisual.java
@@ -21,6 +21,7 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+
 package sc.fiji.omevisual;
 
 import org.scijava.command.Command;
@@ -34,7 +35,8 @@ import net.imagej.ImageJ;
 import net.imagej.display.ImageDisplay;
 import sc.fiji.omevisual.gui.MainAppFrame;
 
-@Plugin(type = Command.class, menuPath = "Plugins>Utilities>Visualize OME Metadata")
+@Plugin(type = Command.class,
+	menuPath = "Plugins>Utilities>Visualize OME Metadata")
 public class OMEVisual implements Command {
 
 	@Parameter

--- a/src/main/java/sc/fiji/omevisual/OMEVisual.java
+++ b/src/main/java/sc/fiji/omevisual/OMEVisual.java
@@ -24,15 +24,18 @@
 
 package sc.fiji.omevisual;
 
+import io.scif.services.TranslatorService;
+
+import net.imagej.Dataset;
+import net.imagej.ImageJ;
+import net.imagej.display.ImageDisplay;
+
 import org.scijava.command.Command;
 import org.scijava.log.LogService;
 import org.scijava.plugin.Parameter;
 import org.scijava.plugin.Plugin;
 
 import loci.formats.ome.OMEXMLMetadata;
-import net.imagej.Dataset;
-import net.imagej.ImageJ;
-import net.imagej.display.ImageDisplay;
 import sc.fiji.omevisual.gui.MainAppFrame;
 
 @Plugin(type = Command.class,
@@ -44,6 +47,9 @@ public class OMEVisual implements Command {
 
 	@Parameter
 	private LogService log;
+
+	@Parameter
+	private TranslatorService translatorService;
 
 	@Parameter
 	private ImageDisplay image;
@@ -67,7 +73,7 @@ public class OMEVisual implements Command {
 
 		Dataset data = (Dataset) image.getActiveView().getData();
 
-		OMEXMLMetadata md = OMEUtils.getOMEXMLMetadata(data, ij);
+		OMEXMLMetadata md = OMEUtils.getOMEXMLMetadata(data, ij, translatorService);
 
 		if (md == null) {
 			return;

--- a/src/main/java/sc/fiji/omevisual/gui/MainAppFrame.java
+++ b/src/main/java/sc/fiji/omevisual/gui/MainAppFrame.java
@@ -21,6 +21,7 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+
 package sc.fiji.omevisual.gui;
 
 import java.io.IOException;
@@ -64,15 +65,18 @@ public class MainAppFrame extends JFrame {
 	}
 
 	/**
-	 * Create the JFXPanel that make the link between Swing (IJ) and JavaFX plugin.
+	 * Create the JFXPanel that make the link between Swing (IJ) and JavaFX
+	 * plugin.
 	 */
 	public void init() {
 		this.fxPanel = new JFXPanel();
 		this.add(this.fxPanel);
 		this.setVisible(true);
 
-		// The call to runLater() avoid a mix between JavaFX thread and Swing thread.
+		// The call to runLater() avoid a mix between JavaFX thread and Swing
+		// thread.
 		Platform.runLater(new Runnable() {
+
 			@Override
 			public void run() {
 				initFX(fxPanel);
@@ -85,7 +89,8 @@ public class MainAppFrame extends JFrame {
 		// Init the root layout
 		try {
 			FXMLLoader loader = new FXMLLoader();
-			loader.setLocation(MainAppFrame.class.getResource("/sc/fiji/omevisual/gui/view/RootLayout.fxml"));
+			loader.setLocation(MainAppFrame.class.getResource(
+				"/sc/fiji/omevisual/gui/view/RootLayout.fxml"));
 			AnchorPane rootLayout = (AnchorPane) loader.load();
 
 			// Get the controller and add an ImageJ context to it.
@@ -102,7 +107,8 @@ public class MainAppFrame extends JFrame {
 
 			controller.fill(md);
 
-		} catch (IOException e) {
+		}
+		catch (IOException e) {
 			e.printStackTrace();
 		}
 	}

--- a/src/main/java/sc/fiji/omevisual/gui/model/GenericModel.java
+++ b/src/main/java/sc/fiji/omevisual/gui/model/GenericModel.java
@@ -3,12 +3,12 @@
  * To change this template file, choose Tools | Templates
  * and open the template in the editor.
  */
+
 package sc.fiji.omevisual.gui.model;
 
 import java.util.List;
 
 /**
- *
  * @author Hadrien Mary
  * @param <T>
  */

--- a/src/main/java/sc/fiji/omevisual/gui/model/ImageModel.java
+++ b/src/main/java/sc/fiji/omevisual/gui/model/ImageModel.java
@@ -3,6 +3,7 @@
  * To change this template file, choose Tools | Templates
  * and open the template in the editor.
  */
+
 package sc.fiji.omevisual.gui.model;
 
 import java.util.ArrayList;
@@ -15,7 +16,6 @@ import loci.formats.ome.OMEXMLMetadata;
 import ome.xml.model.primitives.PositiveInteger;
 
 /**
- *
  * @author Hadrien Mary
  */
 public class ImageModel extends GenericModel<TiffDataModel> {
@@ -44,26 +44,34 @@ public class ImageModel extends GenericModel<TiffDataModel> {
 		this.pixelID = md.getPixelsID(imageID);
 
 		if (md.getPixelsPhysicalSizeX(imageID) != null) {
-			this.physicalSizeX = md.getPixelsPhysicalSizeX(imageID).value().floatValue();
-		} else {
+			this.physicalSizeX = md.getPixelsPhysicalSizeX(imageID).value()
+				.floatValue();
+		}
+		else {
 			this.physicalSizeX = -1;
 		}
 
 		if (md.getPixelsPhysicalSizeY(imageID) != null) {
-			this.physicalSizeY = md.getPixelsPhysicalSizeY(imageID).value().floatValue();
-		} else {
+			this.physicalSizeY = md.getPixelsPhysicalSizeY(imageID).value()
+				.floatValue();
+		}
+		else {
 			this.physicalSizeY = -1;
 		}
 
 		if (md.getPixelsPhysicalSizeZ(imageID) != null) {
-			this.physicalSizeZ = md.getPixelsPhysicalSizeZ(imageID).value().floatValue();
-		} else {
+			this.physicalSizeZ = md.getPixelsPhysicalSizeZ(imageID).value()
+				.floatValue();
+		}
+		else {
 			this.physicalSizeZ = -1;
 		}
 
 		if (md.getPixelsTimeIncrement(imageID) != null) {
-			this.timeIncrement = md.getPixelsTimeIncrement(imageID).value().floatValue();
-		} else {
+			this.timeIncrement = md.getPixelsTimeIncrement(imageID).value()
+				.floatValue();
+		}
+		else {
 			this.timeIncrement = -1;
 		}
 
@@ -98,25 +106,29 @@ public class ImageModel extends GenericModel<TiffDataModel> {
 
 		if (this.physicalSizeX > -1) {
 			rows.add(Arrays.asList("Physical Size X", this.physicalSizeX + " µm"));
-		} else {
+		}
+		else {
 			rows.add(Arrays.asList("Physical Size X", ""));
 		}
 
 		if (this.physicalSizeY > -1) {
 			rows.add(Arrays.asList("Physical Size Y", this.physicalSizeY + " µm"));
-		} else {
+		}
+		else {
 			rows.add(Arrays.asList("Physical Size Y", ""));
 		}
 
 		if (this.physicalSizeZ > -1) {
 			rows.add(Arrays.asList("Physical Size Z", this.physicalSizeZ + " µm"));
-		} else {
+		}
+		else {
 			rows.add(Arrays.asList("Physical Size Z", ""));
 		}
 
 		if (this.timeIncrement > -1) {
 			rows.add(Arrays.asList("Time Increment", this.timeIncrement + " s"));
-		} else {
+		}
+		else {
 			rows.add(Arrays.asList("Time Increment", ""));
 		}
 
@@ -129,7 +141,8 @@ public class ImageModel extends GenericModel<TiffDataModel> {
 		for (int i = 0; i < channels.size(); i++) {
 			Map<String, String> channel = channels.get(i);
 			for (Map.Entry<String, String> entry : channel.entrySet()) {
-				rows.add(Arrays.asList("Channel " + Integer.toString(i) + " - " + entry.getKey(), entry.getValue()));
+				rows.add(Arrays.asList("Channel " + Integer.toString(i) + " - " + entry
+					.getKey(), entry.getValue()));
 			}
 		}
 

--- a/src/main/java/sc/fiji/omevisual/gui/model/TiffDataModel.java
+++ b/src/main/java/sc/fiji/omevisual/gui/model/TiffDataModel.java
@@ -3,6 +3,7 @@
  * To change this template file, choose Tools | Templates
  * and open the template in the editor.
  */
+
 package sc.fiji.omevisual.gui.model;
 
 import java.util.ArrayList;
@@ -17,7 +18,6 @@ import net.imagej.axis.AxisType;
 import ome.xml.model.primitives.NonNegativeInteger;
 
 /**
- *
  * @author Hadrien Mary
  */
 public class TiffDataModel extends GenericModel<GenericModel<?>> {
@@ -39,7 +39,9 @@ public class TiffDataModel extends GenericModel<GenericModel<?>> {
 
 	private final ImageModel imageModel;
 
-	public TiffDataModel(int imageID, int tiffDataID, OMEXMLMetadata md, ImageModel imageModel) {
+	public TiffDataModel(int imageID, int tiffDataID, OMEXMLMetadata md,
+		ImageModel imageModel)
+	{
 		this.imageID = imageID;
 		this.tiffDataID = tiffDataID;
 		this.c = md.getTiffDataFirstC(imageID, tiffDataID);
@@ -60,16 +62,20 @@ public class TiffDataModel extends GenericModel<GenericModel<?>> {
 				this.dt = md.getPlaneDeltaT(imageID, tiffDataID).value().floatValue();
 			}
 			if (md.getPlaneExposureTime(imageID, tiffDataID) != null) {
-				this.exposureTime = md.getPlaneExposureTime(imageID, tiffDataID).value().floatValue();
+				this.exposureTime = md.getPlaneExposureTime(imageID, tiffDataID).value()
+					.floatValue();
 			}
 			if (md.getPlanePositionX(imageID, tiffDataID) != null) {
-				this.posX = md.getPlanePositionX(imageID, tiffDataID).value().floatValue();
+				this.posX = md.getPlanePositionX(imageID, tiffDataID).value()
+					.floatValue();
 			}
 			if (md.getPlanePositionY(imageID, tiffDataID) != null) {
-				this.posY = md.getPlanePositionY(imageID, tiffDataID).value().floatValue();
+				this.posY = md.getPlanePositionY(imageID, tiffDataID).value()
+					.floatValue();
 			}
 			if (md.getPlanePositionZ(imageID, tiffDataID) != null) {
-				this.posZ = md.getPlanePositionZ(imageID, tiffDataID).value().floatValue();
+				this.posZ = md.getPlanePositionZ(imageID, tiffDataID).value()
+					.floatValue();
 			}
 		}
 
@@ -98,31 +104,36 @@ public class TiffDataModel extends GenericModel<GenericModel<?>> {
 
 		if (this.dt >= 0) {
 			rows.add(Arrays.asList("dt", this.dt + " ms"));
-		} else {
+		}
+		else {
 			rows.add(Arrays.asList("dt", ""));
 		}
 
 		if (this.exposureTime >= 0) {
 			rows.add(Arrays.asList("Exposure time", this.exposureTime + " ms"));
-		} else {
+		}
+		else {
 			rows.add(Arrays.asList("Exposure time", ""));
 		}
 
 		if (this.posX >= 0) {
 			rows.add(Arrays.asList("Position X", this.posX + " µm"));
-		} else {
+		}
+		else {
 			rows.add(Arrays.asList("Position X", ""));
 		}
 
 		if (this.posY >= 0) {
 			rows.add(Arrays.asList("Position Y", this.posY + " µm"));
-		} else {
+		}
+		else {
 			rows.add(Arrays.asList("Position Y", ""));
 		}
 
 		if (this.posZ >= 0) {
 			rows.add(Arrays.asList("Position Z", this.posZ + " µm"));
-		} else {
+		}
+		else {
 			rows.add(Arrays.asList("Position Z", ""));
 		}
 

--- a/src/main/java/sc/fiji/omevisual/gui/view/RootLayoutController.java
+++ b/src/main/java/sc/fiji/omevisual/gui/view/RootLayoutController.java
@@ -21,6 +21,7 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+
 package sc.fiji.omevisual.gui.view;
 
 import java.net.URL;
@@ -89,16 +90,15 @@ public class RootLayoutController implements Initializable {
 	private ImageDisplay image;
 
 	@Override
-	public void initialize(URL location, ResourceBundle resources) {
-	}
+	public void initialize(URL location, ResourceBundle resources) {}
 
 	public void setContext(Context context) {
 		context.inject(this);
 	}
 
 	/**
-	 * Not the best place to put ImageDisplay in the gui package but it's easier for
-	 * now. I take any ideas to improve this.
+	 * Not the best place to put ImageDisplay in the gui package but it's easier
+	 * for now. I take any ideas to improve this.
 	 *
 	 * @param image
 	 */
@@ -121,7 +121,8 @@ public class RootLayoutController implements Initializable {
 			for (int j = 0; j < md.getTiffDataCount(i); j++) {
 				TiffDataModel dataModel = new TiffDataModel(i, j, md, imageModel);
 				// md.getTiffDataFirstC(i, j),
-				// md.getTiffDataFirstT(i, j), md.getTiffDataFirstZ(i, j), md.getTiffDataIFD(i,
+				// md.getTiffDataFirstT(i, j), md.getTiffDataFirstZ(i, j),
+				// md.getTiffDataIFD(i,
 				// j));
 				TreeItem<GenericModel<?>> dataItem = new TreeItem<>(dataModel);
 				imageItem.getChildren().add(dataItem);
@@ -130,33 +131,34 @@ public class RootLayoutController implements Initializable {
 		}
 
 		// Handle selection in the tree
-		testTree.getSelectionModel().selectedItemProperty()
-				.addListener((ObservableValue obs, Object oldValue, Object newValue) -> {
-					TreeItem<GenericModel<?>> selectedItem = (TreeItem<GenericModel<?>>) newValue;
-					GenericModel<?> model = selectedItem.getValue();
+		testTree.getSelectionModel().selectedItemProperty().addListener((
+			ObservableValue obs, Object oldValue, Object newValue) -> {
+			TreeItem<GenericModel<?>> selectedItem =
+				(TreeItem<GenericModel<?>>) newValue;
+			GenericModel<?> model = selectedItem.getValue();
 
-					if (model instanceof TiffDataModel) {
-						// Display informations relative to TiffData
-						populateTiffDataInformations((TiffDataModel) model);
+			if (model instanceof TiffDataModel) {
+				// Display informations relative to TiffData
+				populateTiffDataInformations((TiffDataModel) model);
 
-					}
+			}
 
-					if (model instanceof ImageModel) {
-						// Clear TiffData informations
-						this.tiffDataTable.getItems().clear();
+			if (model instanceof ImageModel) {
+				// Clear TiffData informations
+				this.tiffDataTable.getItems().clear();
 
-						// Display informations relative to Image
-						populateImageInformations((ImageModel) model);
-					}
+				// Display informations relative to Image
+				populateImageInformations((ImageModel) model);
+			}
 
-					if (syncWithImageBox.isSelected()) {
-						if (model instanceof TiffDataModel) {
-							// Sync the current selected item with image display
-							Map<AxisType, Long> positions = ((TiffDataModel) model).getPosition();
-							OMEUtils.setPosition(this.image, positions, this.convert);
-						}
-					}
-				});
+			if (syncWithImageBox.isSelected()) {
+				if (model instanceof TiffDataModel) {
+					// Sync the current selected item with image display
+					Map<AxisType, Long> positions = ((TiffDataModel) model).getPosition();
+					OMEUtils.setPosition(this.image, positions, this.convert);
+				}
+			}
+		});
 
 	}
 

--- a/src/test/java/sc/fiji/omevisual/tests/Main.java
+++ b/src/test/java/sc/fiji/omevisual/tests/Main.java
@@ -1,3 +1,4 @@
+
 package sc.fiji.omevisual.tests;
 
 import java.io.IOException;
@@ -24,7 +25,9 @@ public class Main {
 
 		// Open image
 		String fpath;
-		fpath = Main.class.getResource("/sc/fiji/omevisual/tests/testdata/multi-channel-4D-series.ome.tif").getPath();
+		fpath = Main.class.getResource(
+			"/sc/fiji/omevisual/tests/testdata/multi-channel-4D-series.ome.tif")
+			.getPath();
 
 		DatasetIOService io = ij.context().getService(DatasetIOService.class);
 		Dataset ds = io.open(fpath);

--- a/src/test/java/sc/fiji/omevisual/tests/Main.java
+++ b/src/test/java/sc/fiji/omevisual/tests/Main.java
@@ -1,11 +1,14 @@
 
 package sc.fiji.omevisual.tests;
 
+import io.scif.config.SCIFIOConfig;
+import io.scif.services.DatasetIOService;
+
 import java.io.IOException;
 
-import io.scif.services.DatasetIOService;
 import net.imagej.Dataset;
 import net.imagej.ImageJ;
+
 import sc.fiji.omevisual.OMEVisual;
 
 public class Main {
@@ -30,7 +33,7 @@ public class Main {
 			.getPath();
 
 		DatasetIOService io = ij.context().getService(DatasetIOService.class);
-		Dataset ds = io.open(fpath);
+		Dataset ds = io.open(fpath, new SCIFIOConfig().checkerSetOpen(true));
 		ij.display().createDisplay(ds);
 
 		ij.log().info("Load filaments.tif data.");

